src/redux/books/Books.js

```diff
@@ -29,4 +36,57 @@ export const removing = (id) => ({
   id,
 });
 
+export const fetching = (payload) => ({
+
+  type: FETCHBOOKNEW,
+  payload,
+
+});
+
+export const addBooksToApi = (newBook) => async (dispatch) => {
+  axios.post(apiUrl, newBook,
+    {
+      header: {
+        'content-type': 'application/json',
+      },
+    })
+    .then(() => {
+      dispatch(adding(newBook));
+    });
+};
+
+export const fetchBooks = () => async (dispatch) => {
+  const getBooks = axios.get(apiUrl);
+  const response = (await getBooks).data;
+  const bookList = Object.entries(response);
+  const arr = [];
+  bookList.map(([key, value]) => {
+    const obj = {};
+    obj.id = key;
+    obj.title = value[0].title;
+    obj.category = value[0].category;
+    obj.author = value[0].author;
+    return arr.push(obj);
+  });
+
+  dispatch(fetching(arr));
+};
```

Comment:
- [ ] Good job on creating these thunk actions but please use `createAsyncThunk` for creating thunk action creators.

Reply:
Hi, thank you for letting me know that however, I would like to tell you createAsync has some alternatives on their official website.
https://redux.js.org/tutorials/essentials/part-5-async-logic